integration/benchmark/main.go

```diff
@@ -0,0 +1,376 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	flowsdk "github.com/onflow/flow-go-sdk"
+	client "github.com/onflow/flow-go-sdk/access/grpc"
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/integration/utils"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type LoadCase struct {
+	tps      int
+	duration time.Duration
+}
+
+type dataSlice struct {
+	GitSha              string
+	StartTime           time.Time
+	EndTime             time.Time
+	InputTps            float64
+	OutputTps           float64
+	ProStartTransaction float64
+	ProEndTransaction   float64
+}
+
+func main() {
+	sleep := flag.Duration("sleep", 0, "duration to sleep before benchmarking starts")
```

Comment:
a lot of these flags seems like duplicates from integration/loader/main.go.  Can we refactor the shared parameter into a struct and the shared flag setup into a function?

Reply:
This file is intentionally a modified copy of loader/main.go that is used specifically for CI/benchmark related work. We can refactor them both to use some shared code but it is outside the scope of this issue. We can create a new issue and add it to the backlog.

Reply:
deduplicating the code later on will take more effort than doing some simple refactoring now, especially once the code diverge.  let's just fix it now.

I don't mind leaving the flags as is, but as least refactor the big ass loop as the end of main

Reply:
The loop is actually one of the only places that actually is different between the files because of the additional data needed that is then passed forward into BigQuery. If we try to reconverge those we lose the ability to separate out our TPS results by test case. The earlier parts of the file are the ones that are more static and able to be deduped.

Reply:
yes, i noticed that difference and the lack infinite sleep

you can can easily combine the two loop if you refactor calculateTpsSlices & prepareDataForBigQuery into an object, say TPSWriter, and just provide a different implementation in the other loop

Reply:
Rather than try to force a common structure for the two files that will further diverge I have instead removed all extraneous code from the CI implementation that would never be used. We now only have a single load case instead of parsing for multiples which removes the need for looping at all. Additionally many of the flags have been moved to hardcoded values as in a CI environment they will never change. They are still held as variables rather than hardcoded at point of use for ease of maintainability and updating for possible future use cases (e.g. a CI test pointed at testnet rather than localnet).